ocaml/xapi/extauth_plugin_ADpbis.ml

```diff
@@ -18,6 +18,36 @@
 module D = Debug.Make(struct let name="extauth_plugin_ADpbis" end)
 open D
 
+let match_error_tag (lines:string list) =
+  let err_catch_list = 
+    [ "DNS_ERROR_BAD_PACKET",       Auth_signature.E_LOOKUP;
+      "LW_ERROR_PASSWORD_MISMATCH", Auth_signature.E_CREDENTIALS;
+      "LW_ERROR_INVALID_ACCOUNT",   Auth_signature.E_INVALID_ACCOUNT;
+      "LW_ERROR_ACCESS_DENIED",     Auth_signature.E_DENIED;
+      "LW_ERROR_DOMAIN_IS_OFFLINE", Auth_signature.E_UNAVAILABLE;
+      "LW_ERROR_INVALID_OU",       Auth_signature.E_INVALID_OU;
+      (* More errors to be caught here *)
+    ] 
+  in
+  let rec has_err lines err_pattern =
+    match lines with
+    | [] -> false
+    | line :: rest -> 
+      let split_to_words = fun str -> 
```

Comment:
I would move this function before the `match` call to have it upfront instead or redefining it for each element of the list